packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -317,6 +325,54 @@ export class ChatContext {
     }
   }
 
+  /**
+   * Tees any errors encountered in an async function to the state.
+   *
+   * If the function succeeds, clears associated errors from the state.
+   *
+   * @param f Async function to execute.
+   * @param target The error target to tee error to.
+   * @param clearTargets The error targets to clear errors for if the function succeeds. By default, clears errors for `target.
+   * @returns Result of calling `f`. Also re-raises any exceptions thrown from `f`.
+   * @throws ChatError. Exceptions thrown from `f` are tagged with the failed `target.
+   */
+  public async asyncTeeErrorToState<T>(
```

Comment:
nit: shouldn't need the async if the function itself is marked async

Reply:
oh never mind this is a decorator of kinds, never mind keep the async

Reply:
in fact could we use a javascript decorator here? (I've never actually used one so might not be possible but seems like a good application of one)

Reply:
I looked at decorators but can't use one here. They're meant to be used for classes or class methods. We're dispatching a lambda off of a switch statement in an object proxy...

The purpose of decorators is to avoid losing the "class method"-ness of class methods when wrapping with other higher-order functions. Without decorators, the setup would look like:
- You have a class with a method.
- You have a wannaBeDecorator() which accepts a function and returns a modified function.
- If you directly wrap the class method with wannaBeDecorator, you'll lose certain metadata that the function has from being a class method.

The decorator form of wannaBeDecorator() helps preserve that metadata.

In this case, no such need exists -- the functions being modified are already unnamed function objects.

Reference: https://www.simplethread.com/understanding-js-decorators/